### PR TITLE
[DISCO-2045] Investigate K6 load testing in merino-py CD

### PR DESCRIPTION
**Pull request by Trinaa**
# Description
* add an ADR Locust vs k6; Merino-py Performance Test Framework
* 🤦🏻‍♀️ The branch name has a typo, sorry for the confusion

# Issue(s)
[DISCO-2045](https://mozilla-hub.atlassian.net/browse/DISCO-2045)

[DISCO-2045]: https://mozilla-hub.atlassian.net/browse/DISCO-2046?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/adr/0001-locust-vs-k6-merino-py-performance-test-framework.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+* k6 is built by Grafana Labs, inferring easy integration with dashboards
+* k6 has built-in RPS, HTTP request failure and time metrics with customizable URL 
+  break-down
+* k6 supports direct command line usage
+* k6 is feature rich, including built-in functions to generate pass/fail results and 
+  create custom metrics
+
+#### Cons
+
+* The k6 development stack is in Javascript/Typescript. This means:
```
Comment (pdehaan) [inline docs/adr/0001-locust-vs-k6-merino-py-performance-test-framework.md:94]:
```suggestion
* The k6 development stack is in JavaScript/TypeScript. This means:
```

**Review by ncloudioj — APPROVED** [review]
Thanks so much for offering those insights!

I agree with you that overall Locust fits the bill well for Merino with various additional benefits such as shared tooling & code reuse. Let's stick with Locust and build more test scenarios on the existing foundation. Meanwhile, I think we can also invest in related tooling to enhance its usability.

**Comment by taddes** [comment]
Great stuff @Trinaa , well written, good links to valuable context and easy to follow the justification. Thanks for exploring this for the team! 👍

File: docs/adr/0001-locust-vs-k6-merino-py-performance-test-framework.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+migration of Merino from Rust to Python, performance testing was conducted with [k6][2] 
+and focused on the evaluation of request latency. Going forward a unified performance 
+testing solution is preferred, should the test framework be Locust or k6?
+
+## Decision Drivers 
+
+1. The test framework supports the current load test design, a 10-minute test run with 
+   an average load of 1500RPS (see [Merino Load Test Plan][3])
+2. The test framework measures HTTP request failure and client-side latency metrics
+3. The test framework is compatible with the Rapid Release initiative, meaning:
```
Comment (hackebrot) [inline docs/adr/0001-locust-vs-k6-merino-py-performance-test-framework.md:20]:
```suggestion
3. The test framework is compatible with the Rapid Release Model for Firefox Services initiative, meaning:
```

File: docs/adr/0001-locust-vs-k6-merino-py-performance-test-framework.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+* B. k6
+
+## Decision Outcome
+
+Chosen option:
+
+* **A. Locust**
+
+Both k6 and Locust are able to execute the current load test design, report required 
+metrics and fulfill the Rapid Release initiative; However, Locust's Python tech stack
```
Comment (hackebrot) [inline docs/adr/0001-locust-vs-k6-merino-py-performance-test-framework.md:40]: See above

**Review by hackebrot — APPROVED** [review]
LGTM! 🚀

**Pull request merged**